### PR TITLE
refactor: toUpdateDashboard add   adjustWidgetsBeforeSave

### PR DESCRIPTION
**Pull request by xieliuduo**
adjustWidgetsBeforeSave add version

### CODE REVIEW

File: frontend/src/app/pages/DashBoardPage/utils/__tests__/board.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Widget } from '../../pages/Board/slice/types';
+import { adjustWidgetsBeforeSave } from '../board';
+
+describe('board editor events', () => {
+  test('adjustWidgetsBeforeSave', () => {
```
Comment (Cuiyansong) [inline frontend/src/app/pages/DashBoardPage/utils/__tests__/board.test.ts:23]: It is better to give more business info for test description. such as `should get new target version after adjust widgets before save`

File: frontend/src/app/pages/DashBoardPage/utils/board.ts [hunk 2]
```diff
@@ -203,3 +204,23 @@ export const getChartDataView = (views: View[], dataCharts: DataChart[]) => {
   });
   return viewViews;
 };
+
+/**
+ * 
+ * @date 2022-02-09
+ * @param {any} widgets:Widget[]
```
Comment (Cuiyansong) [inline frontend/src/app/pages/DashBoardPage/utils/board.ts:211]: VSCode Plugins `Document This`, should be help to quick create description with JSDoc annotation.

Comment (xieliuduo) [reply frontend/src/app/pages/DashBoardPage/utils/board.ts:211]: good advice

File: frontend/src/app/pages/DashBoardPage/utils/__tests__/board.test.ts (hunk 1 above)
Comment (xieliuduo) [reply frontend/src/app/pages/DashBoardPage/utils/__tests__/board.test.ts:23]: nice

**Pull request merged**